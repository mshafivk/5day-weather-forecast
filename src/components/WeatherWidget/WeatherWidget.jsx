import React from 'react';
import {WeatherTile} from '../WeatherTile/WeatherTile';
import PropTypes from 'prop-types';

export const WeatherWidget = ({forecasts}) =>
  forecasts.length === 0 ? (
    <div data-testid="empty-text">Nothing to display</div>
  ) : (
    forecasts.map(({main, weather, dt_txt}) => (
      <WeatherTile {...main} weather={weather} key={dt_txt} dateTime={dt_txt} />
    ))
  );

WeatherWidget.propTypes = {
  forecasts: PropTypes.arrayOf(
    PropTypes.shape({
      main: PropTypes.shape({
        temp: PropTypes.number,
        humidity: PropTypes.number,
      }).isRequired,
      weather: PropTypes.arrayOf(
        PropTypes.shape({
          main: PropTypes.string,
          description: PropTypes.string,
        }),
      ).isRequired,
      dt_txt: PropTypes.string.isRequired,
    }),
  ),
};
