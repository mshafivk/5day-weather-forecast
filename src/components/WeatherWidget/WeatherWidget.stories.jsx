import React from 'react';
import {WeatherWidget} from './WeatherWidget';

export default {
  title: 'Weather Widget',
  component: WeatherWidget,
};

const Tpl = (args) => <WeatherWidget {...args} />;

export const WeatherWidgetDefault = Tpl.bind({});
WeatherWidgetDefault.args = {
  forecasts: [
    {
      dt_txt: '2020-08-04 18:00:00',
      main: {temp: 234.23, humidity: 84},
      weather: [{main: 'Rain', description: 'light rain'}],
    },
    {
      dt_txt: '2020-08-03 19:00:00',
      main: {temp: 234.23, humidity: 84},
      weather: [{main: 'Rain', description: 'light rain'}],
    },
    {
      dt_txt: '2020-08-02 19:00:00',
      main: {temp: 234.23, humidity: 84},
      weather: [{main: 'Rain', description: 'light rain'}],
    },
  ],
};
WeatherWidgetDefault.storyName = 'Default';

export const EmptyWidget = Tpl.bind({});
EmptyWidget.args = {forecasts: []};
EmptyWidget.storyName = 'When Empty';
