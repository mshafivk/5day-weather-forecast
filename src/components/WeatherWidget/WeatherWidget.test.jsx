import React from 'react';
import {WeatherWidget} from './WeatherWidget';
import {shallow} from 'enzyme';

describe('Weather Widget Component', () => {
  it('Should render properly', () => {
    const props = {
      forecasts: [
        {
          dt_txt: '2020-08-04 18:00:00',
          main: {temp: 234.23, humidity: 84},
          weather: [{main: 'Rain', description: 'light rain'}],
        },
        {
          dt_txt: '2020-08-03 19:00:00',
          main: {temp: 234.23, humidity: 84},
          weather: [{main: 'Rain', description: 'light rain'}],
        },
        {
          dt_txt: '2020-08-02 19:00:00',
          main: {temp: 234.23, humidity: 84},
          weather: [{main: 'Rain', description: 'light rain'}],
        },
      ],
    };
    const wrapper = shallow(<WeatherWidget {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should display a message if no forecasts items are empty ', () => {
    const props = {
      forecasts: [],
    };
    const wrapper = shallow(<WeatherWidget {...props} />);
    expect(wrapper.find('[data-testid="empty-text"]').text()).toEqual(
      'Nothing to display',
    );
  });
});
