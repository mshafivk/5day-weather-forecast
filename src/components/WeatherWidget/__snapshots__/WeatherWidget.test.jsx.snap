// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Widget Component Should render properly 1`] = `
Array [
  <WeatherTile
    dateTime="2020-08-04 18:00:00"
    humidity={84}
    key="2020-08-04 18:00:00"
    temp={234.23}
    weather={
      Array [
        Object {
          "description": "light rain",
          "main": "Rain",
        },
      ]
    }
  />,
  <WeatherTile
    dateTime="2020-08-03 19:00:00"
    humidity={84}
    key="2020-08-03 19:00:00"
    temp={234.23}
    weather={
      Array [
        Object {
          "description": "light rain",
          "main": "Rain",
        },
      ]
    }
  />,
  <WeatherTile
    dateTime="2020-08-02 19:00:00"
    humidity={84}
    key="2020-08-02 19:00:00"
    temp={234.23}
    weather={
      Array [
        Object {
          "description": "light rain",
          "main": "Rain",
        },
      ]
    }
  />,
]
`;
